File: app/src/main/java/com/taibahai/quran/SurahAdapter.java
package com.taibahai.quran;

import static com.network.utils.AppClass.sharedPref;

import android.content.Context;
import android.os.Build;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.taibahai.R;
import com.taibahai.databinding.ItemQuranChapterDetailBinding;

import java.util.ArrayList;
import java.util.List;

public class SurahAdapter extends RecyclerView.Adapter<SurahAdapter.HomeListHolder> {
    private OnItemClickListener listener;
    private List<SurahModel> mData;
    private final Context context;
    LayoutInflater layoutInflater;


    public SurahAdapter(Context context) {
        this.context = context;
        mData = new ArrayList<>();
        layoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    public void updateList(List<SurahModel> tasks) {
        this.mData = tasks;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public HomeListHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_quran_chapter_detail, parent, false);
        return new HomeListHolder(ItemQuranChapterDetailBinding.inflate(layoutInflater, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull HomeListHolder holder, int position) {
        SurahModel model = mData.get(position);
        if (position == mData.size() - 1) {
            holder.binding.footer.setVisibility(View.VISIBLE);
        } else {
            holder.binding.footer.setVisibility(View.GONE);
        }

        holder.binding.ayatArabicText.setText(model.getArabicText());
        holder.binding.ayatEnglishTranslation.setText(model.getEnglishText());
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            holder.binding.ayatEnglishTranslitration.setText(Html.fromHtml(model.getEnglish_translation(), Html.FROM_HTML_MODE_LEGACY));
        } else {
            holder.binding.ayatEnglishTranslitration.setText(Html.fromHtml(model.getEnglish_translation()));
        }
        holder.binding.ayatNumber.setText(String.valueOf(model.getPosition()));


//        holder.binding.shareImage.setOnClickListener(v -> {
//            if (listener != null && position != RecyclerView.NO_POSITION) {
//                listener.onItemClick(holder.itemView, model);
//            }
//        });

        int textSize = sharedPref.getInt(StringUtils.FONT_SIZE);
        if (textSize != 0) {
            holder.binding.ayatArabicText.setTextSize(textSize);
        }

    }

    @Override
    public int getItemCount() {
        if (mData != null)
            return mData.size();
        else return 0;
    }

    public interface OnItemClickListener {
        void onItemClick(View view, SurahModel homeModel);
    }

    public void setOnItemClickListner(OnItemClickListener listener) {
        this.listener = listener;
    }

    static class HomeListHolder extends RecyclerView.ViewHolder {
        ItemQuranChapterDetailBinding binding;

        HomeListHolder(@NonNull ItemQuranChapterDetailBinding itemView) {
            super(itemView.getRoot());
            this.binding = itemView;
        }
    }
}

File: app/src/main/java/com/taibahai/quran/SurahAdapter.kt
package com.taibahai.quran

import android.content.Context
import android.os.Build
import android.text.Html
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.facebook.appevents.codeless.internal.ViewHierarchy.setOnClickListener
import com.network.utils.AppClass
import com.taibahai.R
import com.taibahai.databinding.ItemQuranChapterDetailBinding
import com.taibahai.utils.ShareImage.getBitmapFromView

class SurahAdapter(private val context: Context) :
    RecyclerView.Adapter<SurahAdapter.HomeListHolder>() {
    private var listener: OnItemClickListener? = null
    private var mData: List<SurahModel>?
    var layoutInflater: LayoutInflater

    init {
        mData = ArrayList()
        layoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    fun updateList(tasks: List<SurahModel>?) {
        mData = tasks
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HomeListHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_quran_chapter_detail, parent, false)
        return HomeListHolder(ItemQuranChapterDetailBinding.inflate(layoutInflater, parent, false))
    }

    override fun onBindViewHolder(holder: HomeListHolder, position: Int) {
        val model = mData!![position]
        if (position == mData!!.size - 1) {
            holder.binding.footer.visibility = View.VISIBLE
        } else {
            holder.binding.footer.visibility = View.GONE
        }
        holder.binding.ayatArabicText.text = model.arabicText
        holder.binding.ayatEnglishTranslation.text = model.englishText
        holder.binding.ayatEnglishTranslitration.text =
            Html.fromHtml(model.english_translation, Html.FROM_HTML_MODE_LEGACY)
        holder.binding.ayatNumber.text = model.position.toString()


        holder.binding.ivShare.setOnClickListener {
            context.getBitmapFromView(holder.binding.root)

        }
        val textSize = AppClass.sharedPref.getInt(StringUtils.FONT_SIZE)
        if (textSize != 0) {
            holder.binding.ayatArabicText.textSize = textSize.toFloat()
        }
    }

    override fun getItemCount(): Int {
        return if (mData != null) mData!!.size else 0
    }

    interface OnItemClickListener {
        fun onItemClick(view: View?, homeModel: SurahModel?)
    }

    fun setOnItemClickListner(listener: OnItemClickListener?) {
        this.listener = listener
    }

    class HomeListHolder(var binding: ItemQuranChapterDetailBinding) : RecyclerView.ViewHolder(
        binding.root
    )
}